### ISSUE TITLE
s3 listObjects error: "The data couldn’t be read because it isn’t in the correct format"

### ISSUE THREAD

**Issue by aasoni** [noppoMan/aws-sdk-swift#61]
Hey, I am using the s3 client and am running 
```swift
let request = S3.ListObjectsRequest(bucket: bucketName)
let response = try self.s3.listObjects(request)
```
The second line throws the following error:
**"The data couldn’t be read because it isn’t in the correct format"**

The error looks like it might be from parsing the XML that comes back in the response from S3?

**Comment by cloutiertyler** [comment]
I am getting a similar error when I try to get a single object. Specifically, the `ServiceProtocol` for S3 is specified as `.restxml`:

https://github.com/noppoMan/aws-sdk-swift/blob/master/Sources/AWSSDKSwift/Services/s3/S3_API.swift#L19

Given that that is the case the `AWSClient` tries to parse the response data as XML:

https://github.com/noppoMan/aws-sdk-swift-core/blob/master/Sources/AWSSDKSwiftCore/AWSClient.swift#L453

This fails, however, since the data is just the raw S3 object data and not XML. As far as I can tell this looks like correct behavior for S3: 

http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGET.html

**Comment by cloutiertyler** [comment]
This is the code that I'm using for the get object request.

```swift
let getObjectRequest = S3.GetObjectRequest(bucket: bucket, key: hash)
let getObjectOutput = try s3.getObject(getObjectRequest) // <-- Error is thrown here.
```

**Comment by noppoMan** [comment]
thanks for reporting the issue !
I'll fix it asap.

**Comment by noppoMan** [comment]
@aasoni @cloutiertyler 
I just fixed these bugs by https://github.com/noppoMan/aws-sdk-swift-core/pull/14 and wrote tests for `getObject` and `getObjects` in #63 . It seems to work fine.
I also published version `1.1.2` that includes this fix. I hope it works out.

And thanks again to Tyler for your polite report. it was helpful to me to fix program quickly.
